### PR TITLE
Two pass write invalidate

### PR DESCRIPTION
**Pull request by arutk**
Signed-off-by: Adam Rutkowski <adam.j.rutkowski@intel.com>

### CODE REVIEW

**Comment by mmichal10** [comment]
retest this please

**Comment by arutk** [comment]
retest this please

**Comment by mmichal10** [comment]
retest this please

File: src/engine/engine_wi.c [hunk 1]
```diff
@@ -18,11 +18,49 @@
...
+			mark it as completed */
+		req->wi_second_pass = true;
+	}
+
+	if (req->wi_second_pass) {
+		req->complete(req, req->error);
+		ocf_req_put(req);
+
+		return 0;
+	}
```
Comment (robertbaldyga) [inline src/engine/engine_wi.c:41]: This is not needed, as we already checked for miss before entering this function.

Comment (arutk) [reply src/engine/engine_wi.c:41]: This check is needed when this function called from IO queue after metadata flush

Comment (arutk) [reply src/engine/engine_wi.c]: hit check is indeed seems to be redundant

**Pull request merged**